### PR TITLE
GEODE-5638: Use cmake to generate the SQLiteCLI plugin project into cli folder

### PR DESCRIPTION
**Pull request by mmartell**

### CODE REVIEW

File: clicache/plugins/SQLiteCLI/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+cmake_minimum_required(VERSION 3.10)
+project(Apache.Geode.Plugins.SQLite CSharp)
+
+# Set the .NET Target Framework (Note: This should match the build for Apache.Geode.)
+set (DOTNET_TARGET_FRAMEWORK_VERSION "v4.5.2")
```
Comment (pivotal-jbarrett) [inline clicache/plugins/SQLiteCLI/CMakeLists.txt]: Why is this set in multiple projects. Why is it independent from the `VS_DOTNET_TARGET_FRAMEWORK_VERSION` variable below?

File: clicache/plugins/SQLiteCLI/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+#foreach(var CMAKE_CURRENT_SOURCE_DIR CMAKE_SOURCE_DIR CMAKE_BINARY_DIR CMAKE_CURRENT_BINARY_DIR)
```
Comment (pivotal-jbarrett) [inline clicache/plugins/SQLiteCLI/CMakeLists.txt]: Delete commented out code.

Comment (mmartell) [reply clicache/plugins/SQLiteCLI/CMakeLists.txt]: Done

File: clicache/src/templates/CMakeLists.txt [hunk 3]
```diff
@@ -16,7 +16,7 @@ cmake_minimum_required(VERSION 3.4)
 project(nativeclient.clicache.templates)
 
 # Set the .NET Target Framework (Note: This should match the build for Apache.Geode.)
-set (DOTNET_TARGET_FRAMEWORK_VERSION "v4.5.2")
+set (DOTNET_TARGET_FRAMEWORK_VERSION ${DOTNET_TARGET_FRAMEWORK_VERSION})
```
Comment (pivotal-jbarrett) [inline clicache/src/templates/CMakeLists.txt]: What is the point of redefining it here to the same value it is already set to?

Comment (mmartell) [reply clicache/src/templates/CMakeLists.txt]: Good catch.

File: CMakeLists.txt [hunk 4]
```diff
@@ -35,6 +35,8 @@ if(CMAKE_GENERATOR MATCHES Win64*)
   endif()
 endif()
 
+set(DOTNET_TARGET_FRAMEWORK_VERSION "4.5.2")
```
[the remark this answers is not in the document]
Comment (mmartell) [reply CMakeLists.txt]: Done.

The CMAKE_DOTNET_TARGET_FRAMEWORK_VERSION property can't be used for the *.csproj.in projects in tests/cli. We created a new ticket (https://issues.apache.org/jira/browse/GEODE-5818) finish porting those to cmake. For now, those projects use the DOTNE_TARGET_FRAMEWORK_VERSION.

File: clicache/src/CMakeLists.txt [hunk 5]
```diff
@@ -309,7 +309,7 @@ set_target_properties(${PROJECT_NAME} PROPERTIES
   VS_GLOBAL_KEYWORD "ManagedCProj"
   VS_GLOBAL_PROJECT_TYPES "{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}"
   VS_GLOBAL_ROOTNAMESPACE ${PROJECT_NAME}
-  VS_DOTNET_TARGET_FRAMEWORK_VERSION "v4.5.2"
+  VS_DOTNET_TARGET_FRAMEWORK_VERSION ${DOTNET_TARGET_FRAMEWORK_VERSION}
```
[the remark this answers is not in the document]
Comment (mmartell) [reply clicache/src/CMakeLists.txt]: Done. Except for the projects that aren't ported to cmake yet.

**Pull request merged**